cmd/agent/main.go

```diff
@@ -43,87 +43,12 @@ func main() {
 	// After this point we can use util_log.Logger and stop using the log package
 	util_log.InitLogger(&cfg.Server)
 	logger := util_log.Logger
-
-	var (
-		promMetrics *prom.Agent
-		lokiLogs    *loki.Loki
-		tempoTraces *tempo.Tempo
-		manager     *integrations.Manager
-	)
-
-	srv, err := server.New(cfg.Server)
-	if err != nil {
-		level.Error(logger).Log("msg", "failed to create server", "err", err)
-		os.Exit(1)
-	}
-
-	if cfg.Prometheus.Enabled {
-		promMetrics, err = prom.New(prometheus.DefaultRegisterer, cfg.Prometheus, logger)
-		if err != nil {
-			level.Error(logger).Log("msg", "failed to create prometheus instance", "err", err)
-			os.Exit(1)
-		}
-
-		// Hook up API paths to the router
-		promMetrics.WireAPI(srv.HTTP)
-		promMetrics.WireGRPC(srv.GRPC)
-	}
-
-	if cfg.Loki.Enabled {
-		lokiLogs, err = loki.New(prometheus.DefaultRegisterer, cfg.Loki, logger)
-		if err != nil {
-			level.Error(logger).Log("msg", "failed to create loki log collection instance", "err", err)
-			os.Exit(1)
-		}
-	}
-
-	if cfg.Tempo.Enabled {
-		tempoTraces, err = tempo.New(prometheus.DefaultRegisterer, cfg.Tempo, cfg.Server.LogLevel)
-		if err != nil {
-			level.Error(logger).Log("msg", "failed to create tempo trace collection instance", "err", err)
-			os.Exit(1)
-		}
-	}
-
-	if cfg.Integrations.Enabled {
-		manager, err = integrations.NewManager(cfg.Integrations, logger, promMetrics.InstanceManager())
-		if err != nil {
-			level.Error(logger).Log("msg", "failed to create integrations manager", "err", err)
-			os.Exit(1)
-		}
-
-		if err := manager.WireAPI(srv.HTTP); err != nil {
-			level.Error(logger).Log("msg", "failed wiring endpoints for integrations", "err", err)
-			os.Exit(1)
-		}
-	}
-
-	srv.HTTP.HandleFunc("/-/healthy", func(w http.ResponseWriter, r *http.Request) {
-		w.WriteHeader(http.StatusOK)
-		fmt.Fprintf(w, "Agent is Healthy.\n")
-	})
-	srv.HTTP.HandleFunc("/-/ready", func(w http.ResponseWriter, r *http.Request) {
-		w.WriteHeader(http.StatusOK)
-		fmt.Fprintf(w, "Agent is Ready.\n")
-	})
-
-	if err := srv.Run(); err != nil {
+	srv := NewAgentServer(logger, cfg)
```

Comment:
I'd prefer to give some whitespace in between setting up the logger and the Agent:

```suggestion

	srv := NewAgentServer(logger, cfg)
```